### PR TITLE
Correct merging of anonymous cart with cart from previously logged in user

### PR DESCRIPTION
**Pull request by jrief**
it also fixes many flake8 issues

### CODE REVIEW

**Comment by rfleschenberg** [comment]
There is a warning from Sphinx, can you fix it?

File: shop/static/shop/css/dialog.scss [hunk 1]
```diff
@@ -42,7 +42,7 @@ button[shop-dialog-proceed] {
 	display: inline-block;
 }
 
-.shop-address-location {
+.shop-address-city {
```
Comment (rfleschenberg) [inline shop/static/shop/css/dialog.scss]: If we change "location" to "city" here, we should also change it a few lines above ("contain the fields zip_code and location as shown").

Comment (jrief) [inline shop/static/shop/css/dialog.scss]: it had to be changed, since that field was renamed in ``shop.models.defaults.AddressModelMixin``.

Comment (rfleschenberg) [inline shop/static/shop/css/dialog.scss]: Ahh, I got confused. I thought this was a change in the docs.

**Pull request merged**